File: 2018 서머윈터/스킬트리/solution.py
def solution(skill, skill_trees):
    comparable = [filter_skill(skill, skill_tree) for skill_tree in skill_trees]
    answer = compare(skill, comparable)
    return len(answer)


def filter_skill(skill, skill_tree):
    return "".join(filter(lambda x: x in set(skill), skill_tree))


def compare(skill, comparable):
    return list(filter(lambda x: skill.startswith(x), comparable))

File: 2018 서머윈터/스킬트리/solution.js
function solution(skill, skillTrees) {
    return skillTrees
        .map(filterComparable(skill))
        .filter(compare(skill))
        .length;
};

const filterComparable = skill => skillTree => {
    return skillTree.split('').filter(eachSkill => skill.includes(eachSkill)).join('');
};

const compare = skill => comparable => {
    return skill.startsWith(comparable);
};
